src/io.rs

```diff
@@ -2,85 +2,126 @@
 
 use core::{fmt, slice};
 use core::fmt::Write;
+use core::ptr;
 
 /// File descriptors
-const STDOUT: usize = 1;
-const STDERR: usize = 2;
+static mut STDOUT: isize = -1;
+static mut STDERR: isize = -1;
 
 /// Host's standard error
 struct Stderr;
 
 /// Host's standard output
 struct Stdout;
 
-fn write_all(fd: usize, mut buffer: &[u8]) {
+pub fn get_stdout() -> isize {
+    // Safe: 32-bit accesses are atomic on ARM
+    unsafe{ptr::read_volatile(&STDOUT)}
+}
+
+pub fn get_stderr() -> isize {
+    // Safe: 32-bit accesses are atomic on ARM
+    unsafe{ptr::read_volatile(&STDERR)}
+}
+
+/// Open stdout and stderr.
+pub fn open_streams() -> Result<(),()> {
```

Comment:
Alternatively this could return a `(Stdout, Stderr)` pair in its `Ok` variant.

Or, `Stdout` and `Stderr` could each have an `open` constructor that does half of what this function does and that returns `Result<Self>`. I prefer this option because you can just open what you need.